File: sample-projects/php/src/Domain/Model/ArmorClass.php
<?php

declare(strict_types=1);

namespace De\Sots\CellarsAndCentaurs\Domain\Model;

use De\Sots\CellarsAndCentaurs\Application\CreatureUtil;

final class ArmorClass
{
    private int $total;

    public function __construct(
        private int $base,
        private int $bonus,
        private string $description = CreatureUtil::STANDARD_ARMOR_CLASS_DESCRIPTION
    ) {
        $this->total = $base + $bonus;
    }

    public function getBase(): int
    {
        return $this->base;
    }

    public function setBase(int $base): void
    {
        $this->base = $base;
        $this->updateTotal();
    }

    public function getBonus(): int
    {
        return $this->bonus;
    }

    public function getTotal(): int
    {
        return $this->total;
    }

    public function getDescription(): string
    {
        return $this->description;
    }

    private function updateTotal(): void
    {
        $this->total = $this->base + $this->bonus;
    }
}

File: sample-projects/c/src/domain/model/armor_class.c
#include "../../../include/de/sots/cellarsandcentaurs/domain/model/armor_class.h"
#include "../../../include/de/sots/cellarsandcentaurs/application/application.h"

ArmorClass armor_class_new(int base, int bonus)
{
    return armor_class_new_described(base, bonus, CREATURE_UTIL_STANDARD_ARMOR_CLASS_DESCRIPTION);
}

ArmorClass armor_class_new_described(int base, int bonus, const char *description)
{
    ArmorClass armor_class = { base, bonus, base + bonus, description };
    return armor_class;
}

int armor_class_get_total(const ArmorClass *armor_class)
{
    return armor_class->total;
}

void armor_class_set_bonus(ArmorClass *armor_class, int bonus)
{
    armor_class->bonus = bonus;
    armor_class->total = armor_class->base + bonus;
}
